pkg/combustion/kubernetes_test.go

```diff
@@ -821,3 +1511,191 @@ func TestConfigureKubernetes_SuccessfulRKE2ServerWithManifests(t *testing.T) {
+		Version: "v1.30.3+k3s1",
+		Network: image.Network{
+			APIVIP4: "192.168.1.1",
+			APIVIP6: "fd12:3456:789a::21",
+		},
+	}
+
+	serverConfig := map[string]any{
+		"cluster-cidr": "fd12:3456:789b::/48,10.42.0.0/16",
+		"service-cidr": "fd12:3456:789c::/112,10.43.0.0/16",
+	}
+
+	nodeIPScript, err := createNodeIPScript(ctx, serverConfig)
+	require.NoError(t, err)
+	assert.Contains(t, nodeIPScript, "set-node-ip.sh")
+
+	nodeIPScriptPath := filepath.Join(ctx.CombustionDir, setNodeIPScript)
+	b, err := os.ReadFile(nodeIPScriptPath)
+	require.NoError(t, err)
+
+	contents := string(b)
+
+	assert.Contains(t, contents, "IPv4=true")
+	assert.Contains(t, contents, "IPv6=true")
+	assert.Contains(t, contents, "prioritizeIPv6=true")
+	assert.Contains(t, contents, "CONFIG_FILE=\"/etc/rancher/k3s/config.yaml\"")
+}
+
+func TestCreateNodeIPScriptDualstackIPv4PrioRke2(t *testing.T) {
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes = image.Kubernetes{
+		Version: "v1.30.3+rke2r1",
+		Network: image.Network{
+			APIVIP4: "192.168.1.1",
+			APIVIP6: "fd12:3456:789a::21",
+		},
+	}
+
+	serverConfig := map[string]any{
+		"cluster-cidr": "10.42.0.0/16,fd12:3456:789b::/48",
+		"service-cidr": "10.43.0.0/16,fd12:3456:789c::/112",
+	}
+
+	nodeIPScript, err := createNodeIPScript(ctx, serverConfig)
+	require.NoError(t, err)
+	assert.Contains(t, nodeIPScript, "set-node-ip.sh")
+
+	nodeIPScriptPath := filepath.Join(ctx.CombustionDir, setNodeIPScript)
+	b, err := os.ReadFile(nodeIPScriptPath)
+	require.NoError(t, err)
+
+	contents := string(b)
+
+	assert.Contains(t, contents, "IPv4=true")
+	assert.Contains(t, contents, "IPv6=true")
+	assert.Contains(t, contents, "prioritizeIPv6=false")
+	assert.Contains(t, contents, "CONFIG_FILE=\"/etc/rancher/rke2/config.yaml\"")
+}
+
+func TestCreateNodeIPScriptIPv4Only(t *testing.T) {
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes = image.Kubernetes{
+		Version: "v1.30.3+rke2r1",
+		Network: image.Network{
+			APIVIP4: "192.168.1.1",
+		},
+	}
+
+	serverConfig := map[string]any{}
+
+	nodeIPScript, err := createNodeIPScript(ctx, serverConfig)
+	require.NoError(t, err)
+	assert.Contains(t, nodeIPScript, "")
+}
+
+func TestCreateNodeIPScriptIPv6Only(t *testing.T) {
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes = image.Kubernetes{
+		Version: "v1.30.3+rke2r1",
+		Network: image.Network{
+			APIVIP6: "fd12:3456:789a::21",
+		},
+	}
+
+	serverConfig := map[string]any{
+		"cluster-cidr": "fd12:3456:789b::/48",
+		"service-cidr": "fd12:3456:789c::/112",
+	}
+
+	nodeIPScript, err := createNodeIPScript(ctx, serverConfig)
+	require.NoError(t, err)
+	assert.Contains(t, nodeIPScript, "set-node-ip.sh")
+
+	nodeIPScriptPath := filepath.Join(ctx.CombustionDir, setNodeIPScript)
+	b, err := os.ReadFile(nodeIPScriptPath)
+	require.NoError(t, err)
+
+	contents := string(b)
+
+	assert.Contains(t, contents, "IPv6=true")
```

Comment:
This value is always true. In fact it's obsolete to include in the first place and this also applies to the other cases. `IPv4` is what we always want to assert because it's templated.